src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,612 @@
+	* }
+    */
+	/* constructor(app, stats, logger, data) {
+		super(app, stats, logger, data);
+	} */
+
+	/**
+	 * @see {Syncer#respondsTo}
+	 */
+	static respondsTo(target) {
+		return target === 'webuntis';
+	}
+
+	static params(params, data = {}) {
+		if (!params || !data) {
+			return false;
+		}
+
+		const query = (params || {}).query || {}; // object containing url, username, password, and schoolname
+
+		const validQuery = (
+			query.username != '' &&
+			query.url != '' &&
+			query.password != '' &&
+			query.schoolname != ''
+		);
+		
+		const validData = (
+			['inclusive', 'exclusive'].includes(data.data.type) || (!data.type && !data.data.courseMetadataIds) &&
+			data.datasourceId != ''
+		);
+
+		if (!validData || !validQuery) {
+			return false;
+		}
+		
+		return [{
+			userId: params.userId,
+			webuntisConfig: {
+				url: query.url,
+				schoolname: query.schoolname,
+				username: query.username,
+				password: query.password,
+			},
+			datasourceId: data.datasourceId,
+			datatype: data.data.type,
+			courseMetadataIds: data.data.courseMetadataIds,
+			dryrun: !("dryrun" in params) || (params.dryrun !== 'false' && params.dryrun !== false),
+		}];
+	}
+
+	/**
+	* @see {Syncer#steps}
+    *
+	* Perform either fetching from WebUntis and storing metadata into 'webuntisMetadata' or
+	* Creating actual Schul-Cloud 'courses'.
+	*/
+	steps() {
+		this.logInfo(`Running WebUntis School Year Sync.\n`);
+		// this.logInfo(`Parameters `, this.data);
+
+		if (this.data.dryrun) {
+			return this.createMetaDataFromWebUntisSteps(this.data);
+		} else {
+			return this.createCoursesFromMetaDataAndWebUntisSteps(this.data);
+		}
+	}
+
+	/**
+	* 
+	*/
+	async createMetaDataFromWebUntisSteps(params) {
+		this.stats.success = false;
+		
+		const config = params.webuntisConfig;
+
+		await this.login(config.url, config.schoolname, config.username, config.password);
+		const metaData = await this.fetchMetaData(params);
+		await this.logout();
+		await this.storeMetadata(metaData, params);
+
+		this.stats.success = true;
+	}
+
+	/**
+	* Note: Login to WebUntis is currently not required as all Metadata is fetched
+	* during the first step
+	*/
+	async createCoursesFromMetaDataAndWebUntisSteps(params) {
+		this.stats.success = false;
+
+		const config = params.webuntisConfig;
+
+		const metaData = await this.obtainMetadata(params);
+		// await this.login(config.url, config.schoolname, config.username, config.password);
+		
+		const data = await this.fetchData(metaData);
+
+		// await this.logout();
+
+		await this.createCourses(config, data, params);
+
+		this.stats.success = true;
+	}
+
+	/**
+	* 
+	* @param {*} params 
+	*/
+	getImportConditionEvaluator(params) {
+		if (params.datatype === 'inclusive') {
+			if (!params.courseMetadataIds) {
+				return ((id) => false);
+			}
+			return ((id) => params.courseMetadataIds.includes(id));
+		}
+		if (params.datatype === 'exclusive') {
+			if (!params.courseMetadataIds) {
+				return ((id) => false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['id' is defined but never used. Allowed unused args must match /app|req|res|next|options|params/u.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)